### ISSUE TITLE
Task 1: Image search

### ISSUE THREAD

**Issue by parvashah-create** [parvashah-create/image-search#1]
# Description
You plan to use the approach discussed in https://github.com/matsui528/sis to
implement a search for a dataset from the
https://mmlab.ie.cuhk.edu.hk/projects/DeepFashion.html site. Develop an
application using Streamlit that searches objects based on uploaded images. You
can use 100 images to keep the problem small enough.

# Steps
1. Install the necessary libraries: You will need to install Streamlit, PyTorch, Torchvision, NumPy, and Pillow. You can install them using pip.

2. Download the DeepFashion dataset: Download the dataset from https://mmlab.ie.cuhk.edu.hk/projects/DeepFashion.html and extract it to a folder.

3. Download the pre-trained model: Download the pre-trained model from https://drive.google.com/file/d/1QFRs_MsT6T_3UNrSJdDfJbW8XZKZXU6b/view and extract it to a folder.

4. Load the model: Load the pre-trained model into your application using PyTorch.

5. Create a Streamlit interface: Use Streamlit to create a user interface where users can upload an image and search for similar images in the DeepFashion dataset.

6. Preprocess the uploaded image: Preprocess the uploaded image by resizing it to the size required by the pre-trained model and normalizing it.

7. Encode the image: Use the pre-trained model to encode the image into a feature vector.

8. Search for similar images: Search the DeepFashion dataset for images that are similar to the uploaded image based on the encoded feature vector. You can use the approach discussed in the https://github.com/matsui528/sis repository to perform the search.

9. Display the results: Display the similar images found in the DeepFashion dataset to the user in the Streamlit interface.

10. Deploy the application: Deploy the application to a server or platform where it can be accessed by users.

**Comment by parvashah-create** [comment]
Task 1 completed and pushed
